### PR TITLE
Parse requirements from the setup file

### PR DESCRIPTION
**Pull request by madkinsz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Closes https://github.com/PrefectHQ/prefect/pull/2905

This updates the `meta.yaml` file to pull requirements from `setup.py`. Unfortunately a couple requirements have different names and the required formatting is different so there's some clunky jinja2 string parsing. 

This is passing build locally but I do not have conda installed to sort out the linting yet.

### CODE REVIEW

**Comment by madkinsz** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -50,6 +90,41 @@ test:
...
     - prefect.tasks.github
+  requirements:
+    run:
+      {{ list_requirements(sdata['extras_require']['test']) | indent(4) }}
+
+output:
+  # The main build
+  - name: prefect
+  # The development build
+  - name: prefect-dev
```
Comment (jcrist) [inline recipe/meta.yaml:101]: You can drop the dev build, that extra is more for local development of prefect and CI than anything a user should need to do.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,6 +1,64 @@
...
 
+
+
+
+# Maps dependency names in pip to their conda alternative
+{%
+  set name_mapping =
+  {
+    "msgpack": "msgpack-python",
+    "docker": "docker-py",
```
Comment (jcrist) [inline recipe/meta.yaml:13]: These aren't the only deps that should be special cased, `dask-core` should be used instead of `dask`. There might be more.

**Review by jcrist — COMMENTED** [review]
Interesting, can you expand on why you went with this approach? Its always nice to automate things when possible, but there is a complication cost to automation - the static non-jinja'd version was much easier to understand (though it required some human-in-the-loop).

File: recipe/meta.yaml (hunk 2 above)
Comment (madkinsz) [reply recipe/meta.yaml:13]: Of course :(

**Comment by madkinsz** [comment]
> Interesting, can you expand on why you went with this approach? Its always nice to automate things when possible, but there is a complication cost to automation - the static non-jinja'd version was much easier to understand (though it required some human-in-the-loop).

Yeah this is basically just doing what was suggested in https://github.com/PrefectHQ/prefect/pull/2905#issuecomment-652704756 as a PoC. I agree the complication/automation cost may not be worth it as this isn't as easy to read and I'm honestly surprised at how hard it was to use jinja to process the requirements into what we want. That said, this would reduce the chance for issues when we update the requirements in the future and having it be fully dependent on the `setup.py` as our single source of truth makes sense. I think either way we should probably close the originating PR in prefect since it seems like we don't want the conda recipe in that repo.

**Comment by madkinsz** [comment]
It also seems feasible that we could adjust the parsing in our `setup.py` to get the versions in the format required here so the jinja template would be much simpler.

File: recipe/meta.yaml (hunk 2 above)
Comment (madkinsz) [reply recipe/meta.yaml:13]: I've fixed this one but will leave this as a reminder to compare the remaining libs

**Pull request closed without merging**